hg-fast-export.py

```diff
@@ -566,19 +554,21 @@ def bail(parser,opt):
   if options.fn_encoding!=None:
     fn_encoding=options.fn_encoding
 
-  filter_contents=None
+  plugins=[]
+  if options.plugins!=None:
+    plugins=options.plugins
+
   if options.filter_contents!=None:
-    import shlex
-    filter_contents=shlex.split(options.filter_contents)
+    plugins+=['shell_filter_file_contents='+options.filter_contents]
 
   plugins={}
   plugins['commit_message_filters']=[]
   plugins['file_data_filters']=[]
 
-  if options.plugins and options.pluginpath:
```

Comment:
Apart from the filter_contents part, this belongs to the previous _mechanisms_ commit.